File: two_sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i, num in enumerate(nums):
            tmp_lst = [n for n in nums]
            tmp_lst.remove(num)
            
            try:
                pair_num_idx = tmp_lst.index(target - num)
                
                if pair_num_idx >= i:
                    pair_num_idx += 1
                
                return [i, pair_num_idx]
            except ValueError:
                continue

File: two_sum.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.find(target - nums[i]) != seen.end()) {
                vector<int> result = {i, seen.find(target - nums[i])->second};
                return result;
            } else {
                seen[nums[i]] = i;
            }
        }
        
        return vector<int>{1};
    }
};